src/player.cpp

```diff
@@ -2329,17 +2329,82 @@ std::string reload_cost_text(int moves)
 
 std::string melee_cost_text(int moves)
 {
-    return string_format( ngettext( "Melee and thrown attacks cost %+d movement point. ",
-                                    "Melee and thrown attacks cost %+d movement points. ",
+    return string_format( ngettext( "Melee attacks cost %+d movement point. ",
+                                    "Melee attacks cost %+d movement points. ",
                                     moves ),
                           moves );
 }
 
+std::string thrown_cost_text(int moves)
+{
+    return string_format( ngettext( "Throwing costs %+d movement point. ",
+                                    "Throwing costs %+d movement points. ",
+                                    moves ),
+                          moves );
+}
+
+
+std::string item_handling_cost_text(int moves)
+{
+    return string_format( ngettext( "Handling or manipulating items costs %+d movement point.\n",
+                                    "Handling or manipulating items costs %+d movement points.\n",
+                                    moves ),
+                          moves );
+}
+
+std::string melee_stamina_cost_text( double stamina )
+{
+    return string_format( _( "Melee attacks cost %+.1f stamina. " ), stamina );
+}
+
+std::string stamina_regen_cost_text(int moves)
+{
+    return string_format( ngettext( "Stamina regeneration %+d point per turn.\n",
+                                    "Stamina regeneration %+d points per turn.\n",
+                                    ( moves == -1 ? 1 : 2 ) ),
+                          moves );
+}
+
+std::string ranged_penalty_text(double mod)
+{
+    return string_format( _( "Ranged penalty %+.0f. " ), mod );
+}
+
+std::string throwing_bonus_text(int mod)
+{
+    return string_format( _( "Throwing bonus %+d. " ), mod );
+}
+
+
 std::string dodge_skill_text(double mod)
 {
     return string_format( _( "Dodge skill %+.1f. " ), mod );
 }
 
+// Used a pointer just to avoid touching player.h
+float calc_old_fall_damage_for_encumbrance_penalty( player *player, body_part body_part ) {
+    int torso_encumbrance = player->encumb(bp_torso);
+    int leg_l_encumbrance = player->encumb(bp_leg_l);
+    int leg_r_encumbrance = player->encumb(bp_leg_l);
+    if (body_part == bp_torso) { torso_encumbrance = 0; }
+    if (body_part == bp_leg_l) { leg_l_encumbrance = 0; }
+    if (body_part == bp_leg_r) { leg_r_encumbrance = 0; }
+    float previous_fall_damage = 1.0f;
+    float dex_dodge = player->dex_cur / 2 + player->get_skill_level( skill_dodge );
+    dex_dodge -= ( ( ( leg_l_encumbrance + leg_r_encumbrance ) / 2 ) + ( torso_encumbrance / 1 ) ) / 10;
+    dex_dodge = std::max( 0.0f, dex_dodge );
+    previous_fall_damage *= (100.0f - (dex_dodge * 4.0f)) / 100.0f;
+    if( player->has_trait("PARKOUR") ) {
+        previous_fall_damage *= 2.0f / 3.0f;
+    }
+    return previous_fall_damage;
```

Comment:
This approach doesn't seem to be right. You copy-pasted `player::fall_damage_mod()` and made your tweaks. A more appropriate solution could look somewhat like this:
```c++
float player::fall_damage_mod( int torso_encumb, int legs_encumb ) const {
    ...
}

float player::fall_damage_mod() const {
    return fall_damage_mod( encumb( bp_torso ), encumb( bp_leg_l ) + encumb( bp_leg_r ) );
}
```
Or another (better encapsulated) solution:
```c++
float player::fall_damage_mod_except( const std::bitset<num_bp> &excluded_bps ) const {
    ...
    const int torso_encumb = ( excluded_bps.test( mfb( bp_torso ) ) ) ? encumb( bp_torso ) : 0;
    ...
}

float player::fall_damage_mod() const {
    return fall_damage_mod_except( {} );
}
```
Then when you want to exclude legs from evaluation:
```c++
fall_damage_mod_except( mfb( bp_leg_l ) | mfb( bp_leg_r ) );

```